src/components/VerticalNavigation/VerticalNavigation.js

```diff
@@ -0,0 +1,482 @@
+    // TODO onNavigate prop or something?
+    // TODO router?
+    // FIXME or should this just be in the item and not here?
+  }
+
+  render() {
+    const { items, children } = this.props;
+    // Nav items may be passed either as nested VerticalNavigationitem.subItems, or as nested items in a prop.
+    // The items prop will take priority, if present, and must be an array of item objects (not React components).
+    // If the items prop is not present, items must be expressed as VerticalNavigationitem.subItems instead.
+    // Any non-VerticalNavigationitem.subItems will be rendered in the masthead.
+    // TODO move this explanation to the docs? maybe summarize here and reference them there?
+    const childrenArray =
+      children &&
+      React.Children.count(children) > 0 &&
+      React.Children.toArray(children);
+    const itemsFromChildren =
+      childrenArray &&
+      childrenArray.filter(child => child.type === VerticalNavigationItem);
+    // TODO maybe use displayName here instead of type -- TODO see recompose get display name
+    const itemsFromProps =
+      items &&
+      items.length > 0 &&
+      items.map((primaryItem, i) => (
+        <VerticalNavigationItem
+          item={primaryItem}
+          key={`primary_${primaryItem.title}`}
+        >
+          {primaryItem.subItems &&
+            primaryItem.subItems.map((secondaryItem, j) => (
+              <VerticalNavigationSecondaryItem
+                item={secondaryItem}
+                key={`secondary_${secondaryItem.title}`}
+              >
+                {secondaryItem.subItems &&
+                  secondaryItem.subItems.map((tertiaryItem, j) => (
+                    <VerticalNavigationTertiaryItem
+                      item={tertiaryItem}
+                      key={`tertiary_${tertiaryItem.title}`}
+                    />
+                  ))}
+              </VerticalNavigationSecondaryItem>
+            ))}
+        </VerticalNavigationItem>
+      ));
+    const itemComponents = itemsFromProps || itemsFromChildren || [];
+
+    const masthead =
+      childrenArray &&
+      childrenArray.find(child => child.type === VerticalNavigationMasthead);
+    // TODO maybe use displayName here instead of type
+
+    const {
+      hiddenIcons,
+      pinnableMenus,
+      showBadges,
+      forceHidden,
+      hideMasthead,
+      hoverDelay,
+      hideDelay,
+      notificationDrawer /* TODO notification drawer components? */,
+      isMobile,
+      showMobileNav,
+      navCollapsed,
+      pinnedSecondaryNav,
+      pinnedTertiaryNav,
+      hoverSecondaryNav,
+      hoverTertiaryNav,
+      activePath,
+      mobilePath
+    } = this.props;
+
+    const mobileDepth =
+      mobilePath && mobilePath.split('/').filter(s => s !== '').length;
+    const showMobileSecondary = isMobile && mobileDepth >= 1;
+    const showMobileTertiary = isMobile && mobileDepth >= 2;
+
+    return (
+      <NavContextProvider
+        idPath={'/'}
```

Comment:
should this be a defaultProp?

Reply:
This could be a defaultProp on NavContextProvider, but I was passing it explicitly because the only place where the path should be `'/'` is at the root, in the idPath passed to primary items. If it were a default prop and i forgot to wire up the paths correctly in a sub-item's context provider, i might get `'/'` starting somewhere down the tree and wonder why my paths are truncated, rather than just getting an error.

Reply:
a defaultProp on `VerticalNavigation.js` rather @mturley ? `VerticalNavigation.defaultProps = { idPath: '/' }`

Reply:
But you wouldn't ever pass it as a prop into VerticalNavigation. That idPath should only ever be '/' in the top-level provider.  I could have instead made it a defaultProp on VerticalNavigationItem, but i'm not sure how context and defaultProps play together.

Reply:
Ok, I am just trying to map existing understanding from React-Router...this works for now ;)

Reply:
No worries! I might move this value somewhere else, or align it more with react-router later on when we add that support.